File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest00460.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import java.net.*;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest00460 {

    @GetMapping("/BenchmarkTest00460")
    public void BenchmarkTest00460(@RequestHeader("X-Forwarded-For") String xForwardedFor, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String forwardedIp = xForwardedFor != null ? xForwardedFor : "";
        String data = "" + forwardedIp;
        new Socket(data, 80).close();
        response.setContentType("application/json");
        response.getWriter().print("{\"id\":0}");
    }
}

File: Benchmarks/normal/python/flask/testcode/benchmark_test_00460.py
# SPDX-License-Identifier: Apache-2.0
from flask import request, jsonify


def BenchmarkTest00460():
    raw_body = request.get_data(as_text=True)
    prefix = ''
    data = prefix + str(raw_body)
    return jsonify({'status': 'ok'}), 200, {'X-Echo': str(data)}
